conan/tools/cmake/toolchain/blocks.py

```diff
@@ -559,6 +559,35 @@ def context(self):
         }
 
 
+class PkgConfigBlock(Block):
+    template = textwrap.dedent("""
+        {% if pkg_config %}
+        set(PKG_CONFIG_EXECUTABLE {{ pkg_config }} CACHE FILEPATH "pkg-config executable")
+        {% endif %}
+        {% if pkg_config_path %}
+        if (DEFINED ENV{PKG_CONFIG_PATH})
+        set(ENV{PKG_CONFIG_PATH} "{{ pkg_config_path }}$ENV{PKG_CONFIG_PATH}")
+        else()
+        set(ENV{PKG_CONFIG_PATH} "{{ pkg_config_path }}")
+        endif()
+        {% endif %}
+        """)
+
+    def context(self):
+        pkg_config = self._conanfile.conf.get("tools.gnu:pkg_config", check_type=str)
+        if pkg_config:
+            pkg_config = pkg_config.replace("\\", "/")
+        pkg_config_path = self._conanfile.generators_folder
+        if pkg_config_path:
+            # Path sep based on "os" host settings (as it's done
+            # for "fpic" calculation for instance)
+            os_ = self._conanfile.settings.get_safe("os")
+            pathsep = ":" if os_ != "Windows" else ";"
```

Comment:
Shouldn't we have to take the Windows subsystems into account for the path separators?

Reply:
Also if the ``conan_toolchain.cmake`` will run in the "build" machine, it should be using the ``settings_build``. 
You can check the code in ``deduce_subsystem()`` (maybe use the function too):
```python
 if scope.startswith("build"):
        the_os = conanfile.settings_build.get_safe("os")
```

Reply:
I agree with you but what about all the pieces of code checking only if `self._conanfile.settings.get_safe("os") == "Windows"` not only CMakeToolchain, I'm talking about the rest of `conan.tools`. Should we review all of them? Only to check that we're not messing anything.

Examples:
```python
class VCVars:
    # .....

    def generate(self, scope="build"):
        conanfile = self._conanfile
        os_ = conanfile.settings.get_safe("os")
        if os_ != "Windows":
            return
```
```python
class TargetConfigurationTemplate(CMakeDepsFileTemplate):
     # ....

    @property
    def context(self):
        # ....
        is_win = self.conanfile.settings.get_safe("os") == "Windows"
```
Those are some examples.

Reply:
The ``VCVars`` is most likely not an issue, as VC cannot easily be a cross-compiler to another OS, so the ``settings`` should be ok.

The ``is_win`` for CMakeDeps seems related to ``set_interface_link_directories``, which is for "host" Windows systems too, so it is probably ok too.

The ``CMakeToolchain`` has probably some larger likelihood to be used for cross-build